### DISCUSSION TITLE
23 marzo 2023 - esercizio 1

### DISCUSSION THREAD

**Discussion by IonutCicio** [sapienzastudentsnetwork/basi-di-dati-1#35] [Esami]
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/40034429/0aea4b2f-cb37-458f-b9ee-832953425c47)

**Comment by Warcophyr** [comment]
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/133346488/b1d94d5d-0ea1-4768-8e2c-7e0f5f6f70d2)
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/133346488/18fcebc4-8d67-4513-87f8-8962543b7b00)
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/133346488/b840a384-5193-441d-8815-90b1f741b84b)

**Comment by IonutCicio** [comment]
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/40034429/0ebdb74d-019b-47ea-b63b-3fcbc049b303)

**Reply by Warcophyr** [reply]
@CuriousCI tu la 1b come la hai interpretata perché per come hai risolto così trovi le id tratte e nome fermate di arrivo che hanno una  tratta che parte prima delle 6. il mio dubbio se vedi la mia domanda è se il quesito chiede solo le tratte che partono prima delle 6 oppure tutte quelle che hanno almeno una partenza prima delle 6?

**Reply by IonutCicio** [reply]
> le tratte che hanno la prima partenza prima delle 6:00 del mattino

Se c'è almeno una partenza prima delle 6:00 del mattino, vuol dire che quella tratta ha la *prima partenza* prima delle 6:00 del mattino, quindi la considero. Sinceramente pareva troppo semplice anche a me, ma la richiesta è quella quindi penso che questa soluzione sia giusta.

**Comment by gioele-m-l** [comment]
> Ogni suggerimento o correzione di eventuali errori è ben accetto/a

1a)
1. Seleziono le ***Tratte*** coinvolte in ***Orario*** dove _orapart_ è compresa tra le ore 8:00 e le ore 9:00:

$$TratteOre8\\\_9 = \pi_{Tratta.*}(Tratta \bowtie_{\\#id=\\#idt} (\sigma_{orapart \geq 8:00 \land orapart \leq 9:00}(Orario)))$$

2. Seleziono e ridenomino gli attributi delle ***Fermate*** di "Guidonia"  e faccio lo stesso per quelle di "Roma":

$$FermateGuidonia = \rho_{AG ← A, \forall A∈Schema(Fermata)}(\sigma_{Comune="Guidonia"}(Fermata))$$

$$FermateRoma = \rho_{AR ← A, \forall A∈Schema(Fermata)}(\sigma_{Comune="Roma"}(Fermata))$$

3. Il risultato è la proiezione degli _Id_ delle ***TratteOre8_9*** e dei _nomi_ e _vie_ delle ***Fermate*** di partenza e arrivo dei theta join tra le tabelle create:

$$Result = \pi_{TratteOre8\\\_9.\\#Id, FermateGuidonia.nomeG, FermateGuidonia.viaG, FermateRoma.nomeR, FermateRoma.viaR}(FermateGuidonia \bowtie_{FermateGuidonia.\\#idG=TratteOre8\\\_9.\\#idp} TratteOre8\\\_9 \bowtie_{TratteOre8\\\_9.\\#ida=FermateRoma.\\#idR} FermateRoma)$$

1b)
1. Trovo le ***Tratte*** aventi almeno un ***Orario*** con _partenza_ prima delle ore 6:00:

$$TrattaPrima6am = \pi_{Tratta.*}(Tratta \bowtie_{Tratta.\\#id=Orario.\\#idt} (\sigma_{orapart<6:00}(Orario)))$$

2. Il risultato è la proiezione sugli attributi _#id_ di ***TrattaPrima6am*** e _nome_ di ***Fermata*** del theta join tra le due tabelle:

$$Result = \pi_{TrattaPrima6am.\\#id, Fermata.nome}(TrattaPrima6am \bowtie_{TrattaPrima6am.\\#ida=Fermata.\\#id} Fermata)$$
